src/bentoml/models.py

```diff
@@ -102,19 +103,49 @@ def import_model(
 
     Args:
         tag: the tag of the model to export
-        path: can be one of two things:
+        path: can be one of few things:
               * a folder on the local filesystem
               * an `FS URL <https://docs.pyfilesystem.org/en/latest/openers.html>`_, for example :code:`'s3://my_bucket/folder/my_model.bentomodel'`
+              * A string, the *repo id* of a model repo hosted on https://huggingface.co/
         protocol: (expert) The FS protocol to use when exporting. Some example protocols are :code:`'ftp'`, :code:`'s3'`, and :code:`'userdata'`
         user: (expert) the username used for authentication if required, e.g. for FTP
         passwd: (expert) the username used for authentication if required, e.g. for FTP
         params: (expert) a map of parameters to be passed to the FS used for export, e.g. :code:`{'proxy': 'myproxy.net'}` for setting a proxy for FTP
         subpath: (expert) the path inside the FS that the model should be exported to
         _model_store: the model store to save the model to
+        name: The name to give to the model in the BentoML store
 
     Returns:
         Model: the imported model
     """
+    if path.startswith("hf:"):
+        from ._internal.frameworks.diffusers import (
+            import_model as diffusers_import_model,
+        )
+        from ._internal.frameworks.transformers import (
+            import_model as transformers_import_model,
+        )
+
+        FRAMEWORK_MAPPING = {
+            "transformers": transformers_import_model,
+            "diffusers": diffusers_import_model,
+        }
+
+        _, repo_name = path.split(":")
+        name = name if name is not None else repo_name.replace("/", "--")
+        try:
+            from huggingface_hub import repo_info
+
+            res = repo_info(repo_name)
+            for tag in res.tags:
+                if tag in FRAMEWORK_MAPPING:
+                    return FRAMEWORK_MAPPING[tag](name, repo_name)
```

Comment:
Ugh do we know for sure that all models from `repo_info` will always return tags that contains `transformers` and `diffusers`?

See https://huggingface.co/docs/huggingface_hub/v0.18.0.rc0/en/package_reference/hf_api#huggingface_hub.hf_api.ModelInfo tags can be optional. So this is won't work for some cases.

Which is ok? probably we will need a warning after the iteration saying that it might not be a diffusers/transformers model

```python
for tag in res.tags:
  if tag in {'transformers', 'diffusers'}: return FRAMEWORK_AMPPING[tag](name, repo_name)
else:
  logger.warning("BentoML cannot automatically determine which framework to save the model (transformers, diffusers). Make sure to set the environment BENTOML_IMPORT_FRAMEWORK='transformers|diffusers' to explicitly set loading framework.
```

@larme what do you think about the envvar `BENTOML_IMPORT_FRAMEWORK`?

Reply:
@aarnphm Why not add the framework variable to the function's arguments list? Then let cli interface convert the env var to the function argument?

Reply:
yes that works as well.